pkg/pool/sampler_test.go

```diff
@@ -38,3 +42,67 @@ func TestSamplerStability(t *testing.T) {
 		require.Equal(t, tc.expected, res, "probabilities: %v", tc.probabilities)
 	}
 }
+
+type netmapMock struct {
+	client.Client
+	name string
+	err  error
+}
+
+func newNetmapMock(name string, needErr bool) netmapMock {
+	var err error
+	if needErr {
+		err = fmt.Errorf("not available")
+	}
+	return netmapMock{name: name, err: err}
+}
+
+func (n netmapMock) EndpointInfo(_ context.Context, _ ...client.CallOption) (*client.EndpointInfo, error) {
+	return nil, n.err
+}
+
+func TestHealthyReweight(t *testing.T) {
+	var (
+		weights           = []float64{0.9, 0.1}
+		names             = []string{"node0", "node1"}
+		rebalanceInterval = 100 * time.Millisecond
+		sleepInterval     = 1000 * time.Millisecond
```

Comment:
In rare cases the test may fail. What way can we avoid it?

Reply:
`require.Eventually`?

Reply:
Is it stable now?

Reply:
Seems to be yes